test/framework/controlplane_helpers.go

```diff
@@ -383,8 +383,9 @@ func ScaleAndWaitControlPlane(ctx context.Context, input ScaleAndWaitControlPlan
 
 	patchHelper, err := patch.NewHelper(input.ControlPlane, input.ClusterProxy.GetClient())
 	Expect(err).ToNot(HaveOccurred())
+	scaleBefore := pointer.Int32Deref(input.ControlPlane.Spec.Replicas, 0)
```

Comment:
```suggestion
	scaleBefore := "<unset>"
	if input.ControlPlane.Spec.Replicas != nil {
		scaleBefore = strconv.Itoa(int(*input.ControlPlane.Spec.Replicas))
	}
```
Would be an alternative. Not sure if there's a more straightforward way to render an int32 pointer to string.